apollo-router/src/spec/query.rs

```diff
@@ -388,7 +405,8 @@ impl Query {
                     input,
                     output,
                     path,
-                    field_type,
+                    parent_type,
```

Comment:
My change: this call is recursing from `Int!` to `Int`, it's not entering a new field, so the parent type remains the same.
Before this change you could get an error message like `Invalid value found for field Int!.field`, which doesn't make sense

Reply:
so this is not a huge issue because `parent_type` is only used for that error message, but I worry that this would break assumptions when we change the code later. Could we move it back to `field_type`, but use `parent_type.inner_named_type()` in the error message?

Reply:
The code should assume that the parent type is the type that the selection is executed on, while the field type is the type that the selection returns. `inner_named_type()` is also incorrect as it would mention `Int.someNumber` in the error, when `Int` is the return type, not the type that has the field `someNumber`.

Before this change `parent_type` would sometimes contain the parent type, but sometimes *not* contain the parent type at all. It could be two entirely different things that must be used differently. So I don't think it could possibly be correct before